jellyfin-api/src/jvmMain/kotlin/org/jellyfin/sdk/api/sockets/OkHttpWebsocketSession.kt

```diff
@@ -97,15 +88,28 @@ public class OkHttpWebsocketSession(
 		state.value = SocketInstanceState.CONNECTING
 		webSocket = client.newWebSocket(request, listener)
 
-		messageForwardJob = coroutineScope.launch {
-			for (message in outgoingMessageChannel) {
-				logger.info { "Sending (raw) message $message" }
+		return state.first { it != SocketInstanceState.CONNECTING } == SocketInstanceState.CONNECTED
+	}
 
-				webSocket?.send(message)
-			}
+	override suspend fun send(message: String): Boolean {
+		logger.info { "Sending (raw) message $message" }
+
+		// Invalid state
+		if (state.value != SocketInstanceState.CONNECTED) {
+			logger.warn { "Unable to send message: invalid state (state=${state.value})" }
+			return false
 		}
 
-		return state.first { it != SocketInstanceState.CONNECTING } == SocketInstanceState.CONNECTED
+		// No existing socket
+		if (webSocket == null) {
+			logger.warn { "Unable to send message: webSocket is null" }
+			return false
+		}
+
+		val sent = webSocket!!.send(message)
```

Comment:
Variable & smart cast would be better than using `!!`

Reply:
I can add `val webSocket = webSocket` above the if statement, not sure if that would look better though

Reply:
Or `val socket = webSocket` to prevent using the same name.